metals/src/main/scala/scala/meta/internal/builds/SbtBuildTool.scala

```diff
@@ -51,31 +51,56 @@ case class SbtBuildTool(version: String) extends BuildTool {
           sbtArgs
         ).flatten
     }
-    writeSbtMetalsPlugin(config)
+    removeOldPlugin()
+    writeSbtMetalsPlugin(workspace, config)
     allArgs
   }
+
   override def digest(
       workspace: AbsolutePath
   ): Option[String] = SbtDigest.current(workspace)
   override val minimumVersion: String = "0.13.17"
 
-  private def writeSbtMetalsPlugin(config: MetalsServerConfig): Unit = {
+  private def removeOldPlugin(): Unit = {
+    def pluginsDirectory(version: String): AbsolutePath = {
+      AbsolutePath(System.getProperty("user.home"))
+        .resolve(".sbt")
+        .resolve(version)
+        .resolve("plugins")
+    }
     val plugins =
-      if (version.startsWith("0.13")) SbtBuildTool.pluginsDirectory("0.13")
-      else SbtBuildTool.pluginsDirectory("1.0")
-    plugins.createDirectories()
+      if (version.startsWith("0.13")) pluginsDirectory("0.13")
+      else pluginsDirectory("1.0")
+
+    val metalsFile = plugins.resolve("metals.sbt")
+    if (metalsFile.exists) metalsFile.delete()
+  }
+
+  private def writeSbtMetalsPlugin(
+      workspace: AbsolutePath,
+      config: MetalsServerConfig
+  ): Unit = {
     val bytes = SbtBuildTool
-      .globalMetalsSbt(config.bloopSbtVersion)
+      .sbtPlugin(config.bloopSbtVersion)
       .getBytes(StandardCharsets.UTF_8)
-    val destination = plugins.resolve("metals.sbt")
-    if (destination.isFile && destination.readAllBytes.sameElements(bytes)) {
+    val projectDir = workspace.resolve("project")
+    projectDir.toFile.mkdir()
+    val metalsPluginfile = projectDir.resolve("metals.sbt")
+    if (metalsPluginfile.isFile && metalsPluginfile.readAllBytes.sameElements(
```

Comment:
could you extract the condition to a variable? alternatively you could extract it to an extension method hasContent or something like that.

Reply:
Extracted to a variable.